src/OrleansLoggerProvider/OrleansLoggingDecorator.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using Microsoft.Extensions.Logging;
+
+namespace Orleans.Runtime
+{
+
+    /// <summary>
+    /// Config for message bulking feature
+    /// </summary>
+    public class MessageBulkingConfig
+    {
+        /// <summary>
+        /// Count limit for bulk message output.
+        /// If the same log code is written more than <c>BulkMessageLimit</c> times in the <c>BulkMessageInterval</c> time period, 
+        /// then only the first <c>BulkMessageLimit</c> individual messages will be written, plus a count of how bulk messages suppressed.
+        /// </summary>
+        public int BulkMessageLimit { get; set; } = DefaultBulkMessageLimit;
+
+        /// <summary>
+        /// Default bulk message limit. 
+        /// </summary>
+        public const int DefaultBulkMessageLimit = 5;
+
+        /// <summary>
+        /// Time limit for bulk message output.
+        /// If the same log code is written more than <c>BulkMessageLimit</c> times in the <c>BulkMessageInterval</c> time period, 
+        /// then only the first <c>BulkMessageLimit</c> individual messages will be written, plus a count of how bulk messages suppressed.
+        /// </summary>
+        public TimeSpan BulkMessageInterval { get; set; } = DefaultBulkMessageInterval;
+
+        /// <summary>
+        /// Default bulk message interval
+        /// </summary>
+        public static readonly TimeSpan DefaultBulkMessageInterval = TimeSpan.FromMinutes(1);
+    }
+
+    /// <summary>
+    /// OrleansLoggingDecorator class. User can plug in their own ILogger implementation into this decorator class to add message bulking feature on top of their logger. 
+    /// Message bulking feature will just log eventId count if the same eventId has appear more than BulkMessageLimit in a certain BulkMessageInterval.
+    /// </summary>
+    public class OrleansLoggingDecorator : ILogger
+    {
+        private static readonly int[] excludedBulkLogCodes = {
+            0,
+            100000 //internal runtime error code
+        };
+        private const int BulkMessageSummaryOffset = 500000;
+
+        private readonly MessageBulkingConfig messageBulkingConfig;
+        private readonly ConcurrentDictionary<int, int> recentLogMessageCounts = new ConcurrentDictionary<int, int>();
+        private long lastBulkLogMessageFlushTicks = DateTime.MinValue.Ticks;
+        private readonly ILogger thirdPartyLogger;
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="thirdPartyLogger"></param>
+        public OrleansLoggingDecorator(MessageBulkingConfig config, ILogger thirdPartyLogger)
```

Comment:
please rename thirdPartyLogger to `decoratedLogger` everywhere (or `wrappedLogger` if you prefer), as 3rd party isn't correct.

Reply:
will do